src/latexstyleparser.cpp

```diff
@@ -823,29 +811,19 @@ QStringList LatexStyleParser::readPackageTracing(QString fn) const
 	out << "\\end{document}";
 	tf->close();
 
-
-    QProcess myProc(nullptr);
-	//add exec search path
+	ExecProgram execProgram;
+	execProgram.program = texdefDir + "pdflatex";
 	if (!texdefDir.isEmpty()) {
-		QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
-		if (env.value("SHELL") == "/bin/bash") {
-			env.insert("PATH", env.value("PATH") + ":" + texdefDir);
-			myProc.setProcessEnvironment(env);
-		}
+		execProgram.additionalSearchPaths = texdefDir;
 	}
-	QStringList args;
-	QString result;
-	args << "-draftmode" << "-interaction=nonstopmode" << "--disable-installer" << tf->fileName();
-	myProc.setWorkingDirectory(QFileInfo(tf->fileName()).absoluteDir().absolutePath());
-	myProc.start(texdefDir + "pdflatex", args);
-	args.clear();
-	myProc.waitForFinished();
-	if (myProc.exitCode() != 0)
+	execProgram.arguments << "-draftmode" << "-interaction=nonstopmode" << "--disable-installer" << tf->fileName();
+	execProgram.workingDirectory = QFileInfo(tf->fileName()).absoluteDir().absolutePath();
+	execProgram.execAndWait();
+	if (!execProgram.normalRun || (execProgram.exitCode != 0)) {
```

Comment:
like here, see above

Reply:
OK, changed it so that `execAndWait` effectively returns 
`normalRun && (execProgram.exitCode == 0)`
So now we are checking the result when calling `execProgram.execAndWait`